sklearn/metrics/base.py

```diff
@@ -71,10 +71,11 @@ def _average_binary_score(binary_metric, y_true, y_score, average,
     if y_type not in ("binary", "multilabel-indicator"):
         raise ValueError("{0} format is not supported".format(y_type))
 
+    check_consistent_length(y_true, y_score, sample_weight)
```

Comment:
Is this needed?

Reply:
@lesteve Thanks.
It ensures that the shape of sample_weight is [n_samples]. We need this to pass the common test we already have.

Reply:
My question was: is there a good reason you moved this line up?

Reply:
@lesteve Thanks.
We handle two cases here. binary y_true and multilabel-indicator y_true. For both case, we need to ensure the shape of sample_weight. Currently, we do the check after binary y_true is seperately handled (see the if statement). So we may fail to raise error in some cases.
```python
roc_auc_score([1,0,1], [0,0,1], sample_weight=[1,2,3]) # output 0.875
roc_auc_score([1,0,1], [0,0,1], sample_weight=[1,2,3,4]) # fail to raise error, output 0.875

Reply:
That is less than great. I think we should fix it in _binary_clf_curve though by doing
```py
check_consistent_length(y_true, y_score, sample_weight)
```
instead of
```py
check_consistent_length(y_true, y_score)
```